KI and Manual Workaround for Buildah ClusterTask in Pipelines 1.0 (#22520)
openshift/openshift-docs: modules/op-release-notes-1-0.adoc

== Known issues
* If you are upgrading from an older version of {pipelines-title}, you must delete your existing deployments before upgrading to {pipelines-title} version 1.0. To delete an existing deployment, you must first delete Custom Resources and then uninstall the {pipelines-title} Operator. For more details, see the uninstalling {pipelines-title} section.
* Submitting the same `v1alpha1` Tasks more than once results in an error. Use `oc replace` instead of `oc apply` when re-submitting a `v1alpha1` Task.

> * `buildah` ClusterTask does not work on the {pipelines-title} version 1.0.

Reviewer: ```suggestion
* `buildah` ClusterTask does not work on the {pipelines-title} version 1.0 when adding a new user inside a container.
```

Author: How about - The `buildah` ClusterTask does not work when a new user is added to a container.

Reviewer: 👍

+

> During the Operator installation, the `--storage-driver` flag for the `buildah` ClusterTask is set to `default` value which causes the `/etc/containers/storage.conf` settings to be ignored. This results in the failure of `buildah` ClusterTask.

Reviewer: ```suggestion
During the Operator installation, the `--storage-driver` flag for the `buildah` ClusterTask is not set and due use only the default value which does not set the storage-driver properly in certain case. This results in the failure of `buildah` ClusterTask when adding a user that looks like this : 

`
useradd: /etc/passwd.8: lock file already used
useradd: cannot lock /etc/passwd; try again later.
`

```

+
As a workaround, you can manually set the `--storage-driver` flag to `overlay` value in the `buildah-task.yaml` file:
+
. Login to your cluster as a `cluster-admin`:
+
----
$ oc login -u <login> -p <password> https://openshift.example.com:6443
----

> . Open the `buildah-task.yaml` file for editing:

Reviewer: ```suggestion
. Edit the `clustertask/` `buildah` via the `oc edit` editing:
```

Author: How about - . Use the `oc edit` command to edit `buildah` ClusterTask

+
----
oc edit clustertask buildah
----
+

> The `buildah` clustertask YAML file opens in the editor set by your `EDITOR` environment variable.

Reviewer: ```suggestion
The `buildah` clustertask live YAML file opens in the editor set by your `EDITOR` environment variable.
```

Author: How about - The current version of the `buildah` clustertask YAML file opens in the editor set by your `EDITOR` environment variable.

Reviewer: 👍

== Known issues
* If you are upgrading from an older version of {pipelines-title}, you must delete your existing deployments before upgrading to {pipelines-title} version 1.0. To delete an existing deployment, you must first delete Custom Resources and then uninstall the {pipelines-title} Operator. For more details, see the uninstalling {pipelines-title} section.
* Submitting the same `v1alpha1` Tasks more than once results in an error. Use `oc replace` instead of `oc apply` when re-submitting a `v1alpha1` Task.
* The `buildah` ClusterTask does not work when a new user is added to a container.
+

> During Operator installation, the `--storage-driver` flag for the `buildah` ClusterTask is not specified, therefore the flag is set to its default value. In some cases, this may cause the storage-driver to be set incorrectly. When a new user is added, the incorrect storage-driver results in the failure of `buildah` ClusterTask with the following error:

Reviewer 2: When the Operator is installed? (To keep it parallel with 'specified' in the next part of the sentence.)
this may cause/this causes
the `buildah` ClusterTask

> When the Operator is installed, the `--storage-driver` flag for the `buildah` ClusterTask is not specified, therefore the flag is set to its default value. In some cases, this causes the storage-driver to be set incorrectly. When a new user is added, the incorrect storage-driver results in the failure of the `buildah` ClusterTask with the following error:

Reviewer 3: Do you mean /`storage-driver/` to refer to a specific parameter value or a generic reference to a storage driver? This scans like you either need to enclose the term in backticks or remove the hyphen.

+

> [source, bash]

Reviewer 3: Remove this line - we don't use this styling: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks

----
useradd: /etc/passwd.8: lock file already used

> useradd: cannot lock /etc/passwd; try again later.

Reviewer: The good thing with having the error message here is that it would be easily googable....

----
+

> As a workaround, you can manually set the `--storage-driver` flag to `overlay` value in the `buildah-task.yaml` file:

Reviewer 2: you can manually set> remove 'you can'
set the `--storage-driver` flag to `overlay` value>set the `--storage-driver` flag value to `overlay` ?
Can we put the location of the change first,  Eg. As a workaround, in the `buildah-task.yaml` file, manually set the `--storage-driver` flag value to `overlay`: ?

+
. Login to your cluster as a `cluster-admin`:
+
----
$ oc login -u <login> -p <password> https://openshift.example.com:6443
----
. Use the `oc edit` command to edit `buildah` ClusterTask:
+
----
$ oc edit clustertask buildah
----
+

> The current version of the `buildah` clustertask YAML file opens in the editor set by your `EDITOR` environment variable.

Reviewer 2: in the editor set by your `EDITOR` environment variable.>in an editor? For this task, do we need to mention why a particular editor opens up?

Author: I was told to explicitly mention that file will open up in the editor that user has set in the Editor environment variable. Some users may not be aware of it.

. Under the `steps:` field, locate the following `command` field:

> +

Reviewer 2: Can we say: Under the `steps:` field, locate the `command` field and replace it with the following: <code snippet from the next step>? It will probably help reduce the number of steps.Unless there are many `command` fields in which case, this wont work.
`steps:` remove the colon?

Author: There are many command fields in the yaml. I will remove : from `steps:`

----
 command: ['buildah', 'bud', '--format=$(params.FORMAT)', '--tls-verify=$(params.TLSVERIFY)', '--layers', '-f', '$(params.DOCKERFILE)', '-t', '$(resources.outputs.image.url)', '$(params.CONTEXT)']
----

> . Replace the `command` field above with the following:

Reviewer 3: s/above with the following/with the following values:

> ! above adj
Do not use to indicate a relative location in a document, as in “the above restrictions.” Use “previous” or
“preceding.”

+
----
 command: ['buildah', '--storage-driver=overlay', 'bud', '--format=$(params.FORMAT)', '--tls-verify=$(params.TLSVERIFY)', '--no-cache', '-f', '$(params.DOCKERFILE)', '-t', '$(params.IMAGE)', '$(params.CONTEXT)']

> ----

Reviewer 2: Is the difference in the space before `command` on L 102 and 107 deliberate?

Author: removed the space from 102

. Save the file and exit.

+

> Alternatively, you can also modify the `buildah` ClusterTask YAML file directly on the Web Console by navigating to *Pipelines* -> *Cluster Tasks* -> *buildah*. Select *Edit Cluster Task* from the *Actions* menu and replace the `command` field as shown in the procedure above.

Reviewer 3: web console

s/procedure above./previous procedure.
